{
	"id": "ALAS2023-2023-317",
	"advisories": [
		{
			"id": "ALAS2023-2023-317",
			"title": "Amazon Linux 2023 - ALAS2023-2023-317: Important priority package update for python3.11",
			"description": "Package updates are available for Amazon Linux 2023 that fix the following vulnerabilities:\nCVE-2023-41105:\n\tAn issue was discovered in Python 3.11 through 3.11.4. If a path containing '\\0' bytes is passed to os.path.normpath(), the path will be truncated unexpectedly at the first '\\0' byte. There are plausible cases in which an application would have rejected a filename for security reasons in Python 3.10.x or earlier, but that filename is no longer rejected in Python 3.11.x.\n\nCVE-2023-40217:\n\tAn issue was discovered in Python before 3.8.18, 3.9.x before 3.9.18, 3.10.x before 3.10.13, and 3.11.x before 3.11.5. It primarily affects servers (such as HTTP servers) that use TLS client authentication. If a TLS server-side socket is created, receives data into the socket buffer, and then is closed quickly, there is a brief window where the SSLSocket instance will detect the socket as \"not connected\" and won't initiate a handshake, but buffered data will still be readable from the socket buffer. This data will not be authenticated if the server-side TLS peer is expecting client certificate authentication, and is indistinguishable from valid TLS stream data. Data is limited in size to the amount that will fit in the buffer. (The TLS connection cannot directly be used for data exfiltration because the vulnerable code path requires that the connection be closed on initialization of the SSLSocket.)\n",
			"severity": [
				{
					"type": "vendor",
					"source": "linux-security@amazon.com",
					"vendor": "Important"
				}
			],
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "https://alas.aws.amazon.com/AL2023/ALAS-2023-317.html"
				},
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-40217"
				},
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-41105"
				}
			]
		}
	],
	"vulnerabilities": [
		{
			"id": "CVE-2023-40217",
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-40217"
				}
			]
		},
		{
			"id": "CVE-2023-41105",
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-41105"
				}
			]
		}
	],
	"detection": [
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-libs",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-test",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-debugsource",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-debug",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-debuginfo",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-devel",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-tkinter",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11-idle",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		},
		{
			"ecosystem": "amazon:2023",
			"vulnerable": true,
			"package": {
				"name": "python3.11",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.11.2-2.amzn2023.0.10"
					}
				],
				"fixed": [
					"0:3.11.2-2.amzn2023.0.10"
				]
			}
		}
	],
	"data_source": "amazon"
}
